interfaces/builtin/desktop.go

```diff
@@ -0,0 +1,129 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const desktopSummary = `allows access to basic graphical desktop resources`
+
+const desktopBaseDeclarationSlots = `
+  desktop:
+    allow-installation:
+      slot-snap-type:
+        - core
+`
+
+const desktopConnectedPlugAppArmor = `
+# Description: Can access basic graphical desktop resources. To be used with
+# other interfaces (eg, wayland).
+
+#include <abstractions/dbus-strict>
+#include <abstractions/dbus-session-strict>
+
+#include <abstractions/fonts>
+/var/cache/fontconfig/   r,
+/var/cache/fontconfig/** mr,
+
+# subset of gnome abstraction
+/etc/gtk-3.0/settings.ini r,
+owner @{HOME}/.config/gtk-3.0/settings.ini r,
+# Note: this leaks directory names that wouldn't otherwise be known to the snap
+owner @{HOME}/.config/gtk-3.0/bookmarks r,
+
+/usr/share/icons/                          r,
+/usr/share/icons/**                        r,
+/usr/share/icons/*/index.theme             rk,
+/usr/share/pixmaps/                        r,
+/usr/share/pixmaps/**                      r,
+/usr/share/unity/icons/**                  r,
+/usr/share/thumbnailer/icons/**            r,
+/usr/share/themes/**                       r,
+
+# The snapcraft desktop part may look for schema files in various locations, so
+# allow reading system installed schemas.
+/usr/share/glib*/schemas/{,*}              r,
+/usr/share/gnome/glib*/schemas/{,*}        r,
+/usr/share/ubuntu/glib*/schemas/{,*}       r,
+
+# subset of freedesktop.org
+owner @{HOME}/.local/share/mime/**   r,
+owner @{HOME}/.config/user-dirs.dirs r,
+
+# gmenu
+dbus (send)
+     bus=session
+     interface=org.gtk.Actions
+     member=Changed
+     peer=(name=org.freedesktop.DBus, label=unconfined),
+
+# notifications
+dbus (send)
+    bus=session
+    path=/org/freedesktop/Notifications
+    interface=org.freedesktop.Notifications
+    member="{GetCapabilities,GetServerInformation,Notify}"
+    peer=(label=unconfined),
+
+dbus (receive)
+    bus=session
+    path=/org/freedesktop/Notifications
+    interface=org.freedesktop.Notifications
+    member=NotificationClosed
+    peer=(label=unconfined),
+
+# Allow requesting interest in receiving media key events. This tells Gnome
+# settings that our application should be notified when key events we are
+# interested in are pressed, and allows us to receive those events.
+dbus (receive, send)
+  bus=session
+  interface=org.gnome.SettingsDaemon.MediaKeys
+  path=/org/gnome/SettingsDaemon/MediaKeys
+  peer=(label=unconfined),
+dbus (send)
+  bus=session
+  interface=org.freedesktop.DBus.Properties
+  path=/org/gnome/SettingsDaemon/MediaKeys
+  member="Get{,All}"
+  peer=(label=unconfined),
+
+# Allow use of snapd's internal 'xdg-open'
+/usr/bin/xdg-open ixr,
+/usr/share/applications/{,*} r,
+/usr/bin/dbus-send ixr,
+dbus (send)
+    bus=session
+    path=/
+    interface=com.canonical.SafeLauncher
```

Comment:
There's a second interface for this as you probably know. That is `io.snapcraft.SafeLauncher`. It should be added (can be in a follow-up PR)

Reply:
Ok, thanks. I'll add in a folowup